### PR TITLE
feat(secrets): Make non-entropy signatures take precedence over entropy signatures

### PR DESCRIPTION
**Pull request by ChanochShayner**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
Make non-entropy signatures take precedence over entropy signatures.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/secrets/runner.py [hunk 1]
```diff
@@ -72,6 +72,9 @@
     'Twilio API Key': 'CKV_SECRET_18',
     'Hex High Entropy String': 'CKV_SECRET_19'
 }
+
+ENTROPY_CHECK_IDS = ('CKV_SECRET_6', 'CKV_SECRET_19')
```
Comment (omryMen) [inline checkov/secrets/runner.py:76]: we also have CKV_SECRET_80 in the platform

File: checkov/secrets/runner.py [hunk 2]
```diff
@@ -231,12 +234,8 @@ def run(
                     logging.info(
                         f"Removing secret due to UUID filtering: {hashlib.sha256(secret.secret_value.encode('utf-8')).hexdigest()}")
                     continue
-                if secret_key in secrets_duplication:
-                    logging.debug(
-                        f'Secret was filtered - secrets_duplication. line_number {secret.line_number}, check_id {check_id}')
-                    continue
-                else:
-                    secrets_duplication[secret_key] = True
+                if secret_key in secret_records.keys() and secret_records[secret_key].check_id in ENTROPY_CHECK_IDS and check_id not in ENTROPY_CHECK_IDS:
```
Comment (omryMen) [inline checkov/secrets/runner.py:237]: what if the order will be the other way ?

File: checkov/secrets/runner.py [hunk 3]
```diff
@@ -231,12 +234,8 @@ def run(
...
+                    secret_records.pop(secret_key)
```
Comment (bo156) [inline checkov/secrets/runner.py:238]: don't you need to also add it back in the end?

File: checkov/secrets/runner.py (hunk 1 above)
Comment (bo156) [inline checkov/secrets/runner.py:76]: So basically this means that anyone adding a new check which uses entropy has to remember/know that this list exists. Is there a place or something you have to inherit to create a new secret check?
If we have something like that, we can add an API for understanding wether it's an entopy check or not

File: checkov/secrets/runner.py (hunk 3 above)
Comment (ChanochShayner) [reply checkov/secrets/runner.py]: Nope, we just removed the existing and the new will add in this iteration.

File: checkov/secrets/runner.py (hunk 1 above)
Comment (ChanochShayner) [reply checkov/secrets/runner.py]: 1. Don't think that we will add a new entropy.
2. There is no place to inherit from.

Comment (ChanochShayner) [reply checkov/secrets/runner.py]: Makes very sense, but no place to inherit from.

**Review by omryMen — APPROVED** [review]

**Review by lirshindalman — APPROVED** [review]

File: tests/secrets/custom_and_entropy/main.tf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+resource "aws_instance" "a" {
+  test_pass: "z2b7k2cQfzc+yjP2K8cjuQ8uoorHBpEvC+XWhU3Z5+IdrPQYwr991Lj73xfZ+RA2GzC0wTedDTvb1C2NX+3Gpw==",
```
Comment (gruebel) [inline tests/secrets/custom_and_entropy/main.tf:2]:
```suggestion
  test_pass = "z2b7k2cQfzc+yjP2K8cjuQ8uoorHBpEvC+XWhU3Z5+IdrPQYwr991Lj73xfZ+RA2GzC0wTedDTvb1C2NX+3Gpw=="
```
this should handle the TF parsing error

**Pull request merged**